### PR TITLE
Adds an iterator for the min and max views to the priority queue

### PR DESCRIPTION
**Pull request by AmanuelEphrem**
## Description
This pull request adds an iterator to the priority queue so that elements can be traversed in a familiar manner.

```swift
let queue: PriorityQueue = ...

for val in queue.ascending {
  //traverses from min to max
}

for val in queue.descending {
  //traverses from max to min
}
```
## Testing
I've included unit tests for the new functionality.

@AquaGeek

### CODE REVIEW

File: .swiftpm/xcode/xcshareddata/xcschemes/PriorityQueueModule.xcscheme [hunk 1]
```diff
@@ -1,77 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (AquaGeek) [inline .swiftpm/xcode/xcshareddata/xcschemes/PriorityQueueModule.xcscheme]: Why was this file deleted?

File: Sources/PriorityQueueModule/PriorityQueue.swift [hunk 2]
```diff
@@ -361,3 +361,45 @@ extension PriorityQueue {
         }
     }
 }
+
+extension PriorityQueue {
+
+    /// Adds sequence conformance to both the min and max view of the priority queue
+
+    public struct Iterator: Sequence, IteratorProtocol {
```
Comment (AquaGeek) [inline Sources/PriorityQueueModule/PriorityQueue.swift]: Let's move this out into a separate file `PriorityQueue+Sequence.swift`.

**Review by AquaGeek — CHANGES_REQUESTED** [review]

File: .swiftpm/xcode/xcshareddata/xcschemes/PriorityQueueModule.xcscheme (hunk 1 above)
Comment (AmanuelEphrem) [reply .swiftpm/xcode/xcshareddata/xcschemes/PriorityQueueModule.xcscheme]: I am not sure how that file got deleted as I never touched that folder. Next time I'll double check which files have changed before submitting a PR so this doesn't happen again.

File: Sources/PriorityQueueModule/PriorityQueue.swift (hunk 2 above)
Comment (AmanuelEphrem) [reply Sources/PriorityQueueModule/PriorityQueue.swift]: Should we still do that even though the PriorityQueue doesn't **directly** conform to the Sequence protocol? It just provides access to an iterator (which does conform). 

I'm thinking we can place it in a file called `PriorityQueue+Iterator.swift`, as that is more descriptive as to what is exactly going on.

File: .swiftpm/xcode/xcshareddata/xcschemes/PriorityQueueModule.xcscheme (hunk 1 above)
Comment (AquaGeek) [reply .swiftpm/xcode/xcshareddata/xcschemes/PriorityQueueModule.xcscheme]: No worries! Do you know how to revert the deletion?

File: Sources/PriorityQueueModule/PriorityQueue.swift (hunk 2 above)
Comment (AquaGeek) [reply Sources/PriorityQueueModule/PriorityQueue.swift]: Either way works for me. I agree that it could be confusing to call it `+Sequence`, because it doesn't conform directly. If you want to call it `+Iterator` instead I'm fine with that.

File: .swiftpm/xcode/xcshareddata/xcschemes/PriorityQueueModule.xcscheme (hunk 1 above)
Comment (AmanuelEphrem) [reply .swiftpm/xcode/xcshareddata/xcschemes/PriorityQueueModule.xcscheme]: Yes! I used this command format `git checkout HEAD~1 file.txt` to bring back the deleted file.

Comment (AquaGeek) [reply .swiftpm/xcode/xcshareddata/xcschemes/PriorityQueueModule.xcscheme]: Oh, nice! I didn't realize you could use `git checkout HEAD~1` on specific files! (I guess that makes sense — I use `git checkout --` on specific files all the time.) 👍  #TIL

**Review by AquaGeek — APPROVED** [review]

**Comment by AquaGeek** [comment]
Ooops. I rebased when merging, and I had renamed the `removeMin`/`removeMax` to `popMin`/`popMax` which broke the iterators. I fixed that in 071c1286448fd3db8b741512ec8cdd905e22ff02.
